### PR TITLE
nlme engines for linear_reg

### LINKED ISSUE #17: Can I use lme()?
I am super happy to see this package coming to life!  Is there a way to interface with lme() or is it in the pipeline somewhere?  I think I need it to accommodate heterogeneous error variance (which I think lmer() cannot do).

### PR DESCRIPTION
**Pull request by topepo**
Closes #17 

This adds `"lme"` and `"gls"` engines.

### CODE REVIEW

File: man/rmd/linear_reg_gls.Rmd [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+  fit(depr_score ~ week, data = riesby)
+```
+
+The estimated within-subject correlations are the same:
+
+```{r}
+library(ape)
+
+# lme, use ape package:
+lme_within_sub <- varcomp(lme_fit$fit)/sum(varcomp(lme_fit$fit))
```
Comment (hfrick) [inline man/rmd/linear_reg_gls.Rmd:87]: a use case for `extract_fit_engine()`

File: man/rmd/linear_reg_lme.Rmd [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+ - 	J Pinheiro, and D Bates. 2000. _Mixed-effects models in S and S-PLUS_. Springer, New York, NY
+ 
+ - West, K, Band Welch, and A Galecki. 2014. _Linear Mixed Models: A Practical Guide Using Statistical Software_. CRC Press.
+ 
+  - Thorson, J, Minto, C. 2015, Mixed effects: a unifying framework for statistical modelling in fisheries biology. _ICES Journal of Marine Science_, Volume 72, Issue 5, Pages 1245–1256.
+  
+  - Harrison, XA, Donaldson, L, Correa-Cano, ME, Evans, J, Fisher, DN, Goodwin, CED, Robinson, BS, Hodgson, DJ, Inger, R. 2018. _A brief introduction to mixed effects modelling and multi-model inference in ecology_. PeerJ 6:e4794. 
+  
+  - DeBruine LM, Barr DJ. Understanding Mixed-Effects Models Through Data Simulation. 2021. _Advances in Methods and Practices in Psychological Science_.
```
Comment (hfrick) [inline man/rmd/linear_reg_lme.Rmd:69]:
```suggestion
- J Pinheiro, and D Bates. 2000. _Mixed-effects models in S and S-PLUS_. Springer, New York, NY
 
- West, K, Band Welch, and A Galecki. 2014. _Linear Mixed Models: A Practical Guide Using Statistical Software_. CRC Press.
 
- Thorson, J, Minto, C. 2015, Mixed effects: a unifying framework for statistical modelling in fisheries biology. _ICES Journal of Marine Science_, Volume 72, Issue 5, Pages 1245–1256.
  
- Harrison, XA, Donaldson, L, Correa-Cano, ME, Evans, J, Fisher, DN, Goodwin, CED, Robinson, BS, Hodgson, DJ, Inger, R. 2018. _A brief introduction to mixed effects modelling and multi-model inference in ecology_. PeerJ 6:e4794. 
  
- DeBruine LM, Barr DJ. Understanding Mixed-Effects Models Through Data Simulation. 2021. _Advances in Methods and Practices in Psychological Science_.   
```

File: man/rmd/linear_reg_gls.Rmd [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+```{r}
+gls_fit <- 
+  linear_reg() %>% 
+  set_engine("gls", correlation =  corCompSymm(form = ~ 1 | subject)) %>% 
+  fit(depr_score ~ week, data = riesby)
+
+lme_fit <-
+  linear_reg() %>% 
+  set_engine("lme", random =  ~ 1|subject) %>%
```
Comment (hfrick) [inline man/rmd/linear_reg_gls.Rmd:77]:
```suggestion
  set_engine("lme", random =  ~ 1 | subject) %>% 
```

File: man/rmd/linear_reg_gls.Rmd [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+# gls:
+gls_fit %>% tidy()
+```
+
+
+
+## References
+
+ - 	J Pinheiro, and D Bates. 2000. _Mixed-effects models in S and S-PLUS_. Springer, New York, NY
```
Comment (hfrick) [inline man/rmd/linear_reg_gls.Rmd:120]:
```suggestion
- J Pinheiro, and D Bates. 2000. _Mixed-effects models in S and S-PLUS_. Springer, New York, NY
```

**Review by hfrick — APPROVED** [review]
The section about how to use it in workflows on the engine help page contains a few "gotchas". Good that it's there! We might want to take extra care to surface that information well.

File: man/rmd/linear_reg_gls.Rmd (hunk 1 above)
Comment (topepo) [reply man/rmd/linear_reg_gls.Rmd:87]: I have a separate vignette that goes over that. This is a little bit easier since it references the fit twice.

**Pull request merged**